core/src/main/scala/com/rallyhealth/vapors/factfilter/evaluator/InterpretExprAsFunction.scala

```diff
@@ -231,6 +232,33 @@ final class InterpretExprAsFunction[F[_] : Foldable, V, P]
     }
   }
 
+  // TODO: Where to put this?
+  private type Fn[A] = Input[F, V] => (Output[A], List[Eval[P]])
+  private implicit object FnFunctor extends Functor[Fn] with Semigroupal[Fn] {
+    override def map[A, B](fa: Fn[A])(f: A => B): Fn[B] = fa.andThen {
+      case (o, params) => (o.copy(value = f(o.value)), params)
+    }
+    override def product[A, B](
+      fa: Fn[A],
+      fb: Fn[B],
+    ): Fn[(A, B)] = { input =>
+      val (a, aParams) = fa(input)
+      val (b, bParams) = fb(input)
+      // TODO: Product type should remove evidence if any arg has no evidence
+      (Output((a.value, b.value), a.evidence ++ b.evidence), aParams ::: bParams)
+    }
+  }
```

Comment:
BLKR: I need to add unit tests for how to handle evidence.

Reply:
I'm punting on evidence tracking until later, so I think we can ignore this one for now.